cdap-ui/cypress/integration/pipeline.joiner.spec.ts

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License'); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { loginIfRequired, getArtifactsPoll, dataCy } from '../helpers';
+
+let headers = {};
+
+// We don't have preview enabled in distributed mode.
+// TODO(CDAP-16620) To enable preview in distributed mode. Once it is enabled we can remove.
+let skipPreviewTests = false;
+// @ts-ignore "cy.state" is not in the "cy" type
+const getMochaContext = () => cy.state('runnable').ctx;
+const skip = () => {
+  const ctx = getMochaContext();
+  return ctx.skip();
+};
+
+const closeButton = '[data-testid="close-config-popover"]';
+
+// Pipeline and plugin info
+const JOINER_PIPELINE_NAME = `joiner_pipeline_${Date.now()}`;
+
+const input1 = 'File';
+const input2 = 'File2';
+const sourceNode1 = { nodeName: 'File', nodeType: 'batchsource', nodeId: '0' };
```

Comment:
Should we use input1 variable?

Reply:
I've removed input1 and input2 since we can get the same info from sourceNode1 and sourceNode2.